lib/adding_columns.py

```diff
@@ -57,9 +60,33 @@ def add_number_of_films(data:DataFrame)->DataFrame:
     data = data.join(number_of_films, on="nconst", how="left")
     return data
 
+def extract_link_to_image(text: str) -> str:
+    try:
+        first = 'src="'
+        last = '._V1'
+        start = text.rindex(first) + len(first)
+        end = text.rindex(last, start)
+        return text[start: end]
+    except ValueError:
+        return None
+
+def get_link_to_imdb_image(actor_id: str) -> str:
+    url = f'https://www.imdb.com/name/{actor_id}/mediaindex'
+    response = requests.get(url)
+    soup = BeautifulSoup(response.content, 'html.parser')
+    image_info = str(soup.find('img', attrs={'class': 'poster'}))
+    return extract_link_to_image(image_info)
+
+udf_get_link_to_image = udf(get_link_to_imdb_image)
+
+def add_url_to_actor_image(data:DataFrame)->DataFrame:
+    data = data.withColumn("image_url", udf_get_link_to_image(data.nconst))
+    return data
+
 def add_all_columns(data:DataFrame)->DataFrame:
     data = add_number_of_oscars(data)
     data = add_number_of_globes(data)
     data = add_number_of_emmy_awards(data)
     data = add_number_of_films(data)
+    data = add_url_to_actor_image(data)
```

Comment:
Nie tu bym to dokładał. Spodziewam się, że to będą bardzo solidnie długie run'y. Ja bym to dołożył do części związanej z external kaggle data. Daj znać jeśli będziesz potrzebował calla aby to przegadać.

Reply:
Metody znajdujące URL zostały w tym pliku.
Na koniec ostatniej metody w pliku `pyspark_init.py` dodałem fragment dołączający kolumnę z linkami do zdjęć aktorów do wcześniej wygenerowanych danych. 
Pod metodą jest `#TODO` z notatką żeby uruchomić te metody gdy nowe dane będą potrzebne